UploadServlet.java

```diff
@@ -72,86 +114,96 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
+				Files.copy(is, Paths.get(uploadPath + File.separator + fileName), StandardCopyOption.REPLACE_EXISTING);
 				
-				con = DriverManager.getConnection(ul, u, password);
-				
-				String sql = "insert into article(subject, title, highlight, abstracts, author, time, filename, path) values (?,?,?,?,?,?,?,?)";
-				
-				ps = con.prepareStatement(sql);
-				
-				ps.setString(1, subject);
-				ps.setString(2, title);
-				ps.setString(3, highlights);
-				ps.setString(4, abstracts);
-				ps.setString(5, author);
-				ps.setTimestamp(6, time);
-				ps.setString(7, fileName);
-				ps.setString(8, path);
-				
-				int status = ps.executeUpdate();
+				try
+				{
+					Class.forName("com.mysql.jdbc.Driver");
+					
+					String u = "root";
+					String password = "124536";
+					String ul = "jdbc:mysql://localhost:3306/ooad?autoReconnect=true&&useSSL=false";
+					
+					con = DriverManager.getConnection(ul, u, password);
+					
+					String sql = "insert into article(subject, title, highlight, abstracts, author, time, filename, path) values (?,?,?,?,?,?,?,?)";
+					
+					ps = con.prepareStatement(sql);
+					
+					ps.setString(1, subject);
+					ps.setString(2, title);
+					ps.setString(3, highlights);
+					ps.setString(4, abstracts);
+					ps.setString(5, author);
+					ps.setTimestamp(6, time);
+					ps.setString(7, fileName);
+					ps.setString(8, path);
+					
+					int status = ps.executeUpdate();
+					
+					if(status>0)
+					{
+						session.setAttribute("fileName", fileName);
+						String msg = "" +fileName+ " upload successfully";
+						request.setAttribute("msg", msg);
+						request.setAttribute("subject", subject);
+						request.getRequestDispatcher("NewFile.jsp").forward(request, response);
+
+					}
+							
+			}
 				
-				if(status>0)
+				catch(SQLException | ClassNotFoundException e)
 				{
-					session.setAttribute("fileName", fileName);
-					String msg = "" +fileName+ " upload successfully";
-					request.setAttribute("msg", msg);
-					request.setAttribute("subject", subject);
-					request.getRequestDispatcher("NewFile.jsp").forward(request, response);
-//					System.out.println("uploadpath: "+uploadPath);
+					System.out.println(e);
 				}
 				
-			}
-			
-			catch(SQLException | ClassNotFoundException e)
-			{
-				System.out.println(e);
-			}
-			
-			finally
-			{
-				try
+				finally
 				{
-					if(ps != null)
+					try
 					{
-						ps.close();
+						if(ps != null)
+						{
+							ps.close();
+						}
+						
+						if(con!=null)
+						{
+							con.close();
+						}
 					}
 					
-					if(con!=null)
+					catch(SQLException e)
 					{
-						con.close();
+						out.println(e);
 					}
 				}
 				
-				catch(SQLException e)
-				{
-					out.println(e);
-				}
+			}else {// if the uploaded file type is not PDF, show an alert message
+				
+				JOptionPane.showMessageDialog(null, "Unsupported article type!.\nOnly PDF files are supported. ", "Info", JOptionPane.INFORMATION_MESSAGE);

```

Comment:
@hema-001 

`article type` should be` file format`.

Hui

Reply:
OK.